"use not_existing_crate:*;" causes invalid "unresolved imports" compile errors

```
cargo new test3
```
Add any valid crate as dep:
```
[dependencies]
log4rs = "*" 
```

I tried this code:

```
use log4rs;
use not_existing_crate::{ * }; // or use not_existing_crate::*;

fn main() {
    println!("Hello, world!");
}
```

I expected to see only `not_existing_crate` is reported as "resovled imports".

Instead, this happened: 
```
error[E0432]: unresolved imports `not_existing_crate`, `log4rs`
 --> src/main.rs:1:5
  |
1 | use log4rs;
  |     ^^^^^^
2 | use not_existing_crate::{ * };
  |     ^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `not_existing_crate`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `test3`
```

### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.54.0 (a178d0322 2021-07-26)
binary: rustc
commit-hash: a178d0322ce20e33eac124758e837cbd80a6f633
commit-date: 2021-07-26
host: x86_64-unknown-linux-gnu
release: 1.54.0
LLVM version: 12.0.1
```

The same error with rust 1.56.0.

--- a/compiler/rustc_resolve/src/imports.rs
+++ b/compiler/rustc_resolve/src/imports.rs
@@ -720,7 +720,9 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
                     note: Vec::new(),
                     suggestion: None,
                 };
-                errors.push((path, err));
+                if path.contains("::") {
+                    errors.push((path, err))
+                }
             }
         }
 

--- a/src/test/ui/rust-2018/unresolved-asterisk-imports.rs
+++ b/src/test/ui/rust-2018/unresolved-asterisk-imports.rs
@@ -1,0 +1,6 @@
+// edition:2018
+
+use not_existing_crate::*; //~ ERROR unresolved import `not_existing_crate
+use std as foo;
+
+fn main() {}

--- a/src/test/ui/rust-2018/unresolved-asterisk-imports.stderr
+++ b/src/test/ui/rust-2018/unresolved-asterisk-imports.stderr
@@ -1,0 +1,9 @@
+error[E0432]: unresolved import `not_existing_crate`
+  --> $DIR/unresolved-asterisk-imports.rs:3:5
+   |
+LL | use not_existing_crate::*;
+   |     ^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `not_existing_crate`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0432`.

--- a/src/test/ui/unresolved/unresolved-asterisk-imports.rs
+++ b/src/test/ui/unresolved/unresolved-asterisk-imports.rs
@@ -1,0 +1,4 @@
+use not_existing_crate::*; //~ ERROR unresolved import `not_existing_crate
+use std as foo;
+
+fn main() {}

--- a/src/test/ui/unresolved/unresolved-asterisk-imports.stderr
+++ b/src/test/ui/unresolved/unresolved-asterisk-imports.stderr
@@ -1,0 +1,9 @@
+error[E0432]: unresolved import `not_existing_crate`
+  --> $DIR/unresolved-asterisk-imports.rs:1:5
+   |
+LL | use not_existing_crate::*;
+   |     ^^^^^^^^^^^^^^^^^^ maybe a missing crate `not_existing_crate`?
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0432`.

